### PR TITLE
Build a frontend for Faucet

### PR DESCRIPTION
**Pull request by kchulj**
This is a simple frontend interface for the faucet.
The users can enter their address to ask for test coins.

Part of #38

### CODE REVIEW

**Comment by Geod24** [comment]
Load the bootstrap assets from MaxCDN

File: frontend/src/index.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="">
+    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
+    <meta name="generator" content="Hugo 0.83.1">
```
Comment (Geod24) [inline frontend/src/index.html:8]: 👀

File: frontend/src/index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <title>BOSAGORA Faucet</title>
+
+    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/sign-in/">
```
Comment (Geod24) [inline frontend/src/index.html:11]: 👀

File: frontend/src/index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        -moz-user-select: none;
+        user-select: none;
+      }
+
+      @media (min-width: 768px) {
+        .bd-placeholder-img-lg {
+          font-size: 3.5rem;
+        }
+      }
+    </style>
```
Comment (Geod24) [inline frontend/src/index.html:33]: Why the split ?

File: frontend/src/index.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    <!-- Custom styles for this template -->
+    <link href="styles.css" rel="stylesheet">
+  </head>
+
+  <body class="text-center">
+    <main class="form-signin">
+      <img class="mb-4" src="../assets/bosagora.png" margin-left="50px">
+        <form action="http://eu-001.bosagora.io:2766/send" method="post">
```
Comment (Geod24) [inline frontend/src/index.html:42]: Don't refer to eu-001, use the same host

File: source/faucet/main.d [hunk 5]
```diff
@@ -366,6 +369,36 @@ public class Faucet : FaucetAPI
             this.client.putTransaction(tx);
         }
     }
+
+    /// Returns: The path at which the files are located
+    private string getStaticFilePath () const
+    {
+        if (std.file.exists("frontend/src/index.html"))
+        {
+            writeln(std.file.getcwd());
```
Comment (Geod24) [inline source/faucet/main.d:378]: 👀

File: source/faucet/main.d [hunk 6]
```diff
@@ -366,6 +369,36 @@ public class Faucet : FaucetAPI
...
+        settings.port = url.port;
+        auto router = new URLRouter;
+
+        /// Convenience redirect, as users expect that accessing '/' redirect to index.html
+        router.match(HTTPMethod.GET, "/", staticRedirect("/index.html", HTTPStatus.movedPermanently));
+        /// By default, match the underlying files
+        router.match(HTTPMethod.GET, "*", serveStaticFiles(path));
+
+        listenHTTP(settings, router);
+    }
```
Comment (Geod24) [inline source/faucet/main.d:401]: Wait you shouldn't have 2 routers / `listenHTTP` calls.

**Review by Geod24 — CHANGES_REQUESTED** [review]

File: source/faucet/main.d (hunk 6 above)
Comment (kchulj) [reply source/faucet/main.d:401]: Should I move the code to the main? There is a `new` router there.

File: frontend/src/index.html (hunk 1 above)
Comment (kchulj) [reply frontend/src/index.html:8]: I would need to keep the author though right?

File: frontend/src/index.html (hunk 4 above)
Comment (kchulj) [reply frontend/src/index.html]: Fixed to use the same host.

File: source/faucet/main.d (hunk 6 above)
Comment (kchulj) [reply source/faucet/main.d]: Deleted the `start` method and moved the code to `main` due to duplication.

File: frontend/src/index.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="">
```
Comment (Geod24) [inline frontend/src/index.html:6]: Needs a description though, e.g. "Faucet for BOSAGORA TestNet"

File: frontend/src/index.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        -webkit-user-select: none;
+        -moz-user-select: none;
+        user-select: none;
+      }
+      @media (min-width: 768px) {
+        .bd-placeholder-img-lg {
+          font-size: 3.5rem;
+        }
+      }
+    </style>
```
Comment (Geod24) [inline frontend/src/index.html:26]: My question about the split remains

**Review by Geod24 — APPROVED** [review]
Just two things

**Pull request merged**